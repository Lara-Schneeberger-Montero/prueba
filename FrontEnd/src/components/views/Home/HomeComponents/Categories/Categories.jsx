import React from "react";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { StyledCategories } from "./CategoriesStyles";
import Spinner from "../../../../componentes/Spinner/Spinner";

export default function Categories() {
  const [listaCategorias, setListaCategorias] = useState([]);

  useEffect(() => {
    const settings = {
      method: "GET",
    };

    function listarCategorias() {
      fetch("http://localhost:8080/listarCategorias", settings)
        .then((response) => response.json())
        .then((data) => {
          setListaCategorias(data);
          console.log(data);
        })
        .catch((error) => {
          console.log("Error en el listado general");
          console.log(error);
        });
    }
    listarCategorias();
  }, []);
  
  return (
    <StyledCategories data-testid="styleCategorias">
      <div className="container">
        <h2 className="">Categorias</h2>
        <p></p>
        <ul data-testid="cardsCategorias" className="cardsCategoria">
          {(!listaCategorias.length)
            ? <><Spinner /></>
            : listaCategorias.map((categoria, i) => (
                <li data-testid="lista-categorias" key={i} title={categoria.descripcion_categoria}>
                  <Link
                    to={"/filtro/" + categoria.titulo}
                    className="imagen"
                  >
                    <img
                      data-testid="url"
                      src={categoria.url_categoria}
                      alt={categoria.titulo}
                    />
                  </Link>
                  <div className="contenido">
                    <h3 className="titulo" data-testid="titulo">
                      {categoria.titulo}
                    </h3>
                    <p>{categoria.productos.length} Vehiculos</p>
                  </div>
                </li>
              ))}
        </ul>
      </div>
    </StyledCategories>
  );
}
