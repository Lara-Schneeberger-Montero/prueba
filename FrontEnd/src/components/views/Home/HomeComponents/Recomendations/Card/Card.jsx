import React from "react";
import { Link } from "react-router-dom";
import { StyledCard } from "./Card.style";
import ButtonFullBackground from "../../../../../componentes/ButtonFullBackground";
import point from "../../../../../../media/icons/point.svg";
import star from "../../../../../../media/icons/star.svg";
import wifi from "../../../../../../media/icons/wifi.svg";
import gps from "../../../../../../media/icons/gps.svg";

export default function Card(props) {
  const ubicacionMapa = "https://www.google.com/maps/" + props.mapa;
  const url = "/producto/" + props.idBoton;

  const calificacion = () => {
    let cal = props.calificacion;
    let res = 0;

    if (cal === 10) {
      res = "Excelente";
    } else if (cal >= 8) {
      res = "Muy bueno";
    } else if (cal >= 6) {
      res = "Bueno";
    } else if (cal >= 4) {
      res = "Regular";
    } else if (cal >= 2) {
      res = "Malo";
    } else {
      res = "Muy malo";
    }
    return res;
  };

  return (
    <StyledCard>
      <div id={props.id} className="card-container">
        <div className="card-img">
          <img src={props.img} alt={props.titulo} />
          <i class="fas fa-heart"></i>
        </div>
        <div className="card-body">
          <div data-testid="card-cabecera" className="card-cabecera">
            <div>
              <h5 className="card-categories">
                {props.categoria}
                <img src={star} alt="star" />
                <img src={star} alt="star" />
                <img src={star} alt="star" />
                <img src={star} alt="star" />
                <img src={star} alt="star" />
              </h5>
              <h4 className="card-title">{props.titulo}</h4>
            </div>
            <div className="card-title-points">
              <h4>{props.calificacion}</h4>
              <h5 data-testid="calificacion">{calificacion()}</h5>
            </div>
          </div>
          <div data-testid="card-description" className="card-description">
            <div className="card-ubicacion">
              <p>
                <img src={point} alt="point" />
                {props.ubicacion}
                <a href={ubicacionMapa} target="_blank" rel="noreferrer">
                  VER EN EL MAPA
                </a>
              </p>
            </div>
            <div data-testid="card-accesorios" className="card-accesorios">
              <img src={wifi} alt="wifi" />
              <img src={gps} alt="gps" />
            </div>
            <p className="card-text">{props.text}</p>
          </div>
          <Link to={url}>
            <ButtonFullBackground content="Ver mÃ¡s" />
          </Link>
        </div>
      </div>
    </StyledCard>
  );
}
