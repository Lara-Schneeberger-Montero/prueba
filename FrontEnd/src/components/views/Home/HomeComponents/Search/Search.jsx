import React from "react";
import { useState } from "react";
import { Link } from "react-router-dom";
import { DateRangePicker } from "rsuite";
import { StyledSearch } from "./SearchStyles";
import { useMediaQuery } from "react-responsive";

import Select from "./Select/Select"
import "./calendarSearch.css";
import '../../../../../Styles/calendar.css'

// import { StyleCalendarSearch } from "../../../../../Styles/calendar2";


export default function Search() {
  const [city, setCity] = useState("");
  const [nombre_city, setNombreCity] = useState("");
  
  const handlerCity = (city, nombre) => {
    setCity(city);
    setNombreCity(nombre);
  };

  const isMobile = useMediaQuery({ query: "screen and (max-width: 480px)" });
  // const isTablet = useMediaQuery({
  //   query: "screen and (min-width: 481px) and (max-width: 768px)",
  // });

  function redimension() {
    if (isMobile) {
      return (
        <div className="calendarOneMonth">
          <div>
            <DateRangePicker
            
              showOneCalendar
              // aparece un solo calendario
              // open  //calendario estatico
              placeholder="Check in - Check out"
            />
          </div>
        </div>
      );
    } else{
      return (
        <div className="calendarTwoMonth">
          <DateRangePicker
          
            showTwoCalendar
            // open  //calendario estatico
            placeholder="Check in - Check out"
          />
        </div>
      );
    }
    // } else {
    //   return (
    //     <div className="calendarTwoMonth">
    //       <DateRangePicker
            
    //         showTwoCalendar
    //         // readOnly={true}
    //         // open   //calendario estatico
    //         placeholder="Check in - Check out"
    //       />
    //     </div>
    //   );
    // }
  }

  return (
    <StyledSearch data-testid="styleSearch">
      <h2>Encuentra un veh√≠culo</h2>
      <form data-testid="form-search">
        <Select cityChange={handlerCity}/>
        {redimension('resize')}
        {!nombre_city ? (
          <Link to={"/#"} className="buscar" type="submit">
            <p>Buscar</p>
          </Link>
        ) : (
          <Link to={"/filtro/" + nombre_city + "/" + city} className="buscar" type="submit">
            <p>Buscar</p>
          </Link>
        )}
      </form>
    </StyledSearch>
  );
}
