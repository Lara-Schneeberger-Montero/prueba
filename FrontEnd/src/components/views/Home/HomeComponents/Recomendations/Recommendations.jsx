import React, { useEffect, useState } from "react";
import Card from "./Card/Card";
import Spinner from "../../../../componentes/Spinner/Spinner";
import { StyledRecomendations } from "./RecomendationsStyles";


export default function Recommendations() {
  const [carList, setCarList] = useState([]);

  /* ---------------------------- FETCH ---------------------------- */
  useEffect(() => {
    const settings = {
      method: "GET",
    };

    function list() {
      fetch("http://localhost:8080/producto/listarProductos", settings)
        .then((response) => response.json())
        .then((data) => {
          setCarList(data);
          console.log("setListCardlist");
          console.log(data);
        })
        .catch((error) => {
          console.log("holaa");
          console.log(error);
        });
    }
    list();
  }, []);

  return (
    <StyledRecomendations>
      <div className="container">
        <h2>Recomendaciones</h2>
        <div data-testid="Cards" className="cards">
          {(!carList.length) ? (
            <><Spinner /></>
          ) : (
            carList.map((product) => (
              <Card
                idBoton={product.id_producto}
                key={product.id_producto}
                img={product.imagenes[0].url}
                categoria={product.titulo}
                titulo={product.nombre_producto}
                ubicacion={product.ciudad.nombre + ", " + product.ciudad.pais_ciudad}
                text={product.descripcion_producto}
                mapa={"place/"+product.ciudad.nombre}
                //calificacion={product.calificacion}
                calificacion={8}
                id_producto={product.id_producto}
              />
            ))
          )}
        </div>
      </div>
    </StyledRecomendations>
  );
}

// <>
// {carList.length === 0 ? (
//   <> </>
// ) : (
//   carList.map((car, i) => (        ))
//   )}
// </>
