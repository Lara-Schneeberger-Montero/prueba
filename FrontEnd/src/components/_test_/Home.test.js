import { render, screen, cleanup } from '@testing-library/react';
import Categories from '../views/Home/HomeComponents/Categories/Categories';
import Card from '../views/Home/HomeComponents/Recomendations/Card/card'
import { Link } from "react-router-dom";
import { BrowserRouter, Router, Switch, Redirect } from 'react-router-dom';
import Recommendations from '../views/Home/HomeComponents/Recomendations/Recommendations'
import '@testing-library/jest-dom/extend-expect';
import Search from '../views/Home/HomeComponents/Search/Search';
import { DateRangePicker } from 'rsuite';

import { createMemoryHistory } from "history";
const history = createMemoryHistory();

afterEach(cleanup);
describe('Test de home', () => {
    test('Categories', () => {
        const { getByTestId } = render(
            <Router history={history}>
                <Categories />
            </Router>
        );
        expect(getByTestId('styleCategorias')).toBeInTheDocument();
        expect(getByTestId('cardsCategorias')).toBeInTheDocument();
    })

    test('Cards', () => {
        const { getByTestId, getAllByAltText } = render(
            <Router history={history}>
                <Card />
            </Router>
        );
        expect(getByTestId('card-cabecera')).toBeInTheDocument();
        expect(getByTestId('card-description')).toBeInTheDocument();
        expect(getByTestId('card-accesorios')).toBeInTheDocument();
        expect(getAllByAltText('star').length).toBe(5)
    })

    test('Recomendation', () => {
        const data = {
            key: 0,
            img: "https://i.blogs.es/e6e3bc/duramax-ss-650/450_1000.jpg",
            category: "Cl치sicos",
            title: "Carro cl치sico",
            location: "Bogot치",
            description: "El mejor autito del mundo",
            mapa: "place/Bogot치",
            calificacion: "9",
        }
        const { getByTestId } = render(
            <Router history={history}>
                <Recommendations />
            </Router>
        );
        const contenedor = render(
            <Router history={history}>
                <Card {...data} />
            </Router>)

        expect(getByTestId('Cards')).toBeInTheDocument();
        expect(contenedor).toMatchSnapshot();
    })


    test('Search', () => {
        const { getByTestId, getAllByRole, contenedor } = render(
            <Router history={history}>
                <Search />
            </Router>
        );
        expect(getByTestId('styleSearch')).toBeInTheDocument();
        expect(getByTestId('form-search')).toBeInTheDocument();
        // expect(contenedor.toMatchInlineSnapshot(<DateRangePicker className="datePicker"
        // placeholder="Seleccione la fecha"/>)).toBeTruthy();
        expect(getAllByRole('option').length).toBe(5);
    })

})
