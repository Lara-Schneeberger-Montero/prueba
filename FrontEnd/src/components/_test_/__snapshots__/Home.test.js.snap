// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de home Recomendation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-dkPtRN ipAYVi"
      >
        <div
          class="container"
        >
          <h2>
            Recomendaciones
          </h2>
          <div
            class="cards"
            data-testid="Cards"
          >
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Carro clásico"
                    src="https://i.blogs.es/e6e3bc/duramax-ss-650/450_1000.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Clásicos
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Carro clásico
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        9
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Muy bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Bogotá
                        <a
                          href="https://www.google.com/maps/place/Bogotá"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor autito del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Clásico Camaro"
                    src="https://www.autonocion.com/wp-content/uploads/2020/04/Chevrolet-Camaro-e1587570480785.jpg "
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Clásicos
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Clásico Camaro
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        10
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Muy bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Buenos Aires
                        <a
                          href="https://www.google.com/maps/place/Buenos+Aires,+Argentina"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor clásico de la historia del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Deportivo urbano"
                    src="http://www.placervial.com/2/images/wallpapers/g/c84f4dabe77fc31c26b0b018931fd21a.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Deportivos
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Deportivo urbano
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        8
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Muy bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Medellín
                        <a
                          href="https://www.google.com/maps/place/Medellín,+Antioquia"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor deportivo urbano del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Deportivo Honda"
                    src="https://i.blogs.es/c09d77/1366_2000/1366_2000.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Deportivos
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Deportivo Honda
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        6
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Tandil
                        <a
                          href="https://www.google.com/maps/place/Tandil,+Provincia+de+Buenos+Aires,+Argentina"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor Honda deportivo del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Familiar 5 puertas"
                    src="https://autosysaludcom.files.wordpress.com/2018/12/2018-Honda-CR-V.jpg?quality=80&strip=all"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Familiar
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Familiar 5 puertas
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        8
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Muy bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Cartagena
                        <a
                          href="https://www.google.com/maps/place/Cartagena,+Provincia+de+Cartagena,+Bolívar"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor 5 puertas del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="Familiar 7 asientos"
                    src="https://i.blogs.es/b18fcc/suzuki-ertiga-2019/840_560.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        Familiar
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        Familiar 7 asientos
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        7
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Rosario
                        <a
                          href="https://www.google.com/maps/place/Rosario,+Santa+Fe,+Argentina"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor 7 asientos del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="De lujo Ferrari 288 GTO"
                    src="https://static.motor.es/fotos-noticias/2021/03/ferrari-288-gto-nero-dk-engineering-202176004-1615107868_1.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        De lujo
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        De lujo Ferrari 288 GTO
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        9
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Muy bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Cali
                        <a
                          href="https://www.google.com/maps/place/Cali,+Valle+del+Cauca"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor Ferrari del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sc-gsDKAQ fEtXLX"
            >
              <div
                class="card-container"
              >
                <div
                  class="card-img"
                >
                  <img
                    alt="De lujo Lambo"
                    src="https://noticias.coches.com/wp-content/uploads/2016/04/ares-design_lamborghini-huracan-lb724-2015_r6.jpg"
                  />
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="card-cabecera"
                    data-testid="card-cabecera"
                  >
                    <div>
                      <h5
                        class="card-categories"
                      >
                        De lujo
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                        <img
                          alt="star"
                          src="star.svg"
                        />
                      </h5>
                      <h4
                        class="card-title"
                      >
                        De lujo Lambo
                      </h4>
                    </div>
                    <div
                      class="card-title-points"
                    >
                      <h4>
                        6
                      </h4>
                      <h5
                        data-testid="calificacion"
                      >
                        Bueno
                      </h5>
                    </div>
                  </div>
                  <div
                    class="card-description"
                    data-testid="card-description"
                  >
                    <div
                      class="card-ubicacion"
                    >
                      <p>
                        <img
                          alt="point"
                          src="point.svg"
                        />
                        Cordoba
                        <a
                          href="https://www.google.com/maps/place/Córdoba,+Argentina"
                          rel="noreferrer"
                          target="_blank"
                        >
                          VER EN EL MAPA
                        </a>
                      </p>
                    </div>
                    <div
                      class="card-accesorios"
                      data-testid="card-accesorios"
                    >
                      <img
                        alt="wifi"
                        src="wifi.svg"
                      />
                      <img
                        alt="gps"
                        src="gps.svg"
                      />
                    </div>
                    <p
                      class="card-text"
                    >
                      El mejor Lamborghini aventador del mundo
                    </p>
                  </div>
                  <a
                    href="/producto"
                  >
                    <button>
                      Ver más
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div>
      <div
        class="sc-gsDKAQ fEtXLX"
      >
        <div
          class="card-container"
        >
          <div
            class="card-img"
          >
            <img
              src="https://i.blogs.es/e6e3bc/duramax-ss-650/450_1000.jpg"
            />
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-cabecera"
              data-testid="card-cabecera"
            >
              <div>
                <h5
                  class="card-categories"
                >
                  <img
                    alt="star"
                    src="star.svg"
                  />
                  <img
                    alt="star"
                    src="star.svg"
                  />
                  <img
                    alt="star"
                    src="star.svg"
                  />
                  <img
                    alt="star"
                    src="star.svg"
                  />
                  <img
                    alt="star"
                    src="star.svg"
                  />
                </h5>
                <h4
                  class="card-title"
                />
              </div>
              <div
                class="card-title-points"
              >
                <h4>
                  9
                </h4>
                <h5
                  data-testid="calificacion"
                >
                  Muy bueno
                </h5>
              </div>
            </div>
            <div
              class="card-description"
              data-testid="card-description"
            >
              <div
                class="card-ubicacion"
              >
                <p>
                  <img
                    alt="point"
                    src="point.svg"
                  />
                  <a
                    href="https://www.google.com/maps/place/Bogotá"
                    rel="noreferrer"
                    target="_blank"
                  >
                    VER EN EL MAPA
                  </a>
                </p>
              </div>
              <div
                class="card-accesorios"
                data-testid="card-accesorios"
              >
                <img
                  alt="wifi"
                  src="wifi.svg"
                />
                <img
                  alt="gps"
                  src="gps.svg"
                />
              </div>
              <p
                class="card-text"
              />
            </div>
            <a
              href="/producto"
            >
              <button>
                Ver más
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDKAQ fEtXLX"
    >
      <div
        class="card-container"
      >
        <div
          class="card-img"
        >
          <img
            src="https://i.blogs.es/e6e3bc/duramax-ss-650/450_1000.jpg"
          />
        </div>
        <div
          class="card-body"
        >
          <div
            class="card-cabecera"
            data-testid="card-cabecera"
          >
            <div>
              <h5
                class="card-categories"
              >
                <img
                  alt="star"
                  src="star.svg"
                />
                <img
                  alt="star"
                  src="star.svg"
                />
                <img
                  alt="star"
                  src="star.svg"
                />
                <img
                  alt="star"
                  src="star.svg"
                />
                <img
                  alt="star"
                  src="star.svg"
                />
              </h5>
              <h4
                class="card-title"
              />
            </div>
            <div
              class="card-title-points"
            >
              <h4>
                9
              </h4>
              <h5
                data-testid="calificacion"
              >
                Muy bueno
              </h5>
            </div>
          </div>
          <div
            class="card-description"
            data-testid="card-description"
          >
            <div
              class="card-ubicacion"
            >
              <p>
                <img
                  alt="point"
                  src="point.svg"
                />
                <a
                  href="https://www.google.com/maps/place/Bogotá"
                  rel="noreferrer"
                  target="_blank"
                >
                  VER EN EL MAPA
                </a>
              </p>
            </div>
            <div
              class="card-accesorios"
              data-testid="card-accesorios"
            >
              <img
                alt="wifi"
                src="wifi.svg"
              />
              <img
                alt="gps"
                src="gps.svg"
              />
            </div>
            <p
              class="card-text"
            />
          </div>
          <a
            href="/producto"
          >
            <button>
              Ver más
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
